internals-js/src/validate.ts

```diff
@@ -129,13 +129,14 @@ class Validator {
     return this.errors;
   }
 
-  private validateHasType(elt: { type?: Type, coordinate: string, sourceAST?: ASTNode }) {
+  private validateHasType(elt: { type?: Type, coordinate: string, sourceAST?: ASTNode }): boolean {
```

Comment:
Note to reviewers: the changes to this file are completely unrelated to the rest of the patch. It's just that when I was working on this, I run into this and that was easy enough to fix, so I did. 

Namely, this method validates that we don't have some types unset in the schema (which can't really happen when we parse a schema because it's invalid syntax, but if one build a schema programmatically, this could happen), but in a few cases, we were following it by other validations that assumed the type was set. So the change here simply have the method return whether the type is set or not and the other changes in this file simply skip the dependent validation when the type is indeed unset. Concretely, this ensures we get good errors instead of some random "undefined"-access one thrown.